index.bs

```diff
@@ -6809,36 +6809,41 @@ a chunk</dfn> from a {{ReadableStreamDefaultReader}} |reader|, given a [=read re
 
 <div algorithm="read all bytes">
  <p>To <dfn export for="ReadableStreamDefaultReader" lt="read all bytes|reading all bytes">read all
- bytes</dfn> from a {{ReadableStreamDefaultReader}} |reader|, perform the following steps. The
- result will be a {{Promise}} for a [=byte sequence=].
+ bytes as a promise</dfn> from a {{ReadableStreamDefaultReader}} |reader|, perform the following
+ steps. The result will be a {{Promise}} for a [=byte sequence=].
 
  1. Let |promise| be [=a new promise=].
- 1. Let |bytes| be an empty [=byte sequence=].
- 1. [=Read-loop=] given |reader|, |bytes|, and |promise|.
- 1. Return |promise|.
+ 1. Let |success steps| given a [=byte sequence=] |data| be to [=resolve=] |promise| with |data|.
+ 1. Let |failure steps| given a JavaScript value |error| be to [=reject=] |promise| with |error|.
+ 1. [=Consume all bytes=] from |reader| given |success steps| and |failure steps|.
+
+ <p>To <dfn export for="ReadableStreamDefaultReader">consume all bytes</dfn> from
+ {{ReadableStreamDefaultReader}} |reader|, given |success steps|, which is an algorithm accepting a
+ [=byte sequence=], and |failure steps|, which is an algorithm accepting a JavaScript value,
+ [=read-loop=] given |reader|, a new [=byte sequence=], |success steps|, and |failure steps|.
 
  <div algorithm="read-loop">
-  For the purposes of the above algorithm, to <dfn>read-loop</dfn> given |reader|, |bytes|, and
-  |promise|:
+  For the purposes of the above algorithm, to <dfn>read-loop</dfn> given |reader|, |bytes|,
+  |success steps|, and |failure steps|:
 
   1. Let |readRequest| be a new [=read request=] with the following [=struct/items=]:
    : [=read request/chunk steps=], given |chunk|
    ::
-    1. If |chunk| is not a {{Uint8Array}} object, [=reject=] |promise| with a {{TypeError}} and
+    1. If |chunk| is not a {{Uint8Array}} object, call |failure steps| with a {{TypeError}} and
        abort these steps.
     1. Append the bytes represented by |chunk| to |bytes|.
```

Comment:
While here, it might be best to use https://webidl.spec.whatwg.org/#dfn-get-buffer-source-copy to more rigorously convert |chunk| to a byte sequence.

Reply:
Done